tensorflow-framework/src/main/java/org/tensorflow/framework/activations/Activation.java

```diff
@@ -14,24 +14,137 @@
 =======================================================================*/
 package org.tensorflow.framework.activations;
 
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
 import org.tensorflow.Operand;
 import org.tensorflow.op.Ops;
 import org.tensorflow.types.family.TNumber;
 
-/**
- * Interface for Activations
- *
- * @param <T> the data type of the input and the result
- */
+/** Interface for Activations */
 @FunctionalInterface
-public interface Activation<T extends TNumber> {
+public interface Activation {
+
+  /**
+   * Creates an Activation instance based on the name as known to the TensorFlow engine.
+   *
+   * @param name the activation name
+   * @return the Activation
+   * @throws NullPointerException if name is null
+   * @throws IllegalArgumentException if the name is not a known ActivationType
+   */
+  static Activation create(String name) {
+    ActivationType type = ActivationType.of(name);
+    return type.getInstance();
+  }
+
+  /**
+   * Creates an Activation getInstance based on a configuration as produced by TensorFlow.
+   *
+   * @param config a Map object containing the Activation's state. This Map object must contain at
+   *     least a {@code name} key.
+   *     <pre>{@code
+   * "name" : String - this is the TensorFlow Engine's Activation name
+   * }
+   * }</pre>
+   *
+   * @return the Activation
+   * @throws NullPointerException if config is null, or the activation name is missing from the Map.
+   * @throws IllegalArgumentException if the name contained in the config map is not a known
+   *     ActivationType
+   */
+  static Activation create(Map<String, Object> config) {
+    String activationName = (String) config.get("name");
+    ActivationType type = ActivationType.of(activationName);
+    return type.getInstance(config);
+  }
 
   /**
    * Gets the calculation operation for the activation.
    *
    * @param tf the TensorFlow Ops
    * @param input the input tensor
    * @return The operand for the activation
+   * @param <T> the data type of the input and the result
    */
-  Operand<T> call(Ops tf, Operand<T> input);
+  <T extends TNumber> Operand<T> call(Ops tf, Operand<T> input);
+
+  /**
+   * The Enumerations for creating Activations based an activation name, with either an empty
+   * constructor or a constructor that takes a Map object that contains the Activation's state.
+   */
+  enum ActivationType {
```

Comment:
Instead of having `Activation.ActivationType`, let's move this to an outer class called `Activations`

Reply:
OK